sdks/go/pkg/beam/core/runtime/exec/sdf.go

```diff
@@ -313,3 +313,94 @@ func (n *ProcessSizedElementsAndRestrictions) Down(ctx context.Context) error {
 func (n *ProcessSizedElementsAndRestrictions) String() string {
 	return fmt.Sprintf("SDF.ProcessSizedElementsAndRestrictions[%v] Out:%v", path.Base(n.PDo.Fn.Name()), IDs(n.PDo.Out...))
 }
+
+// SdfFallback is an executor used when an SDF isn't expanded into steps by the
+// runner, indicating that the runner doesn't support splitting. It executes all
+// the SDF steps together in one unit.
+type SdfFallback struct {
+	PDo *ParDo
+
+	initRestInv *cirInvoker
+	splitInv    *srInvoker
+	trackerInv  *ctInvoker
+}
+
+// ID just defers to the ParDo's ID method.
+func (n *SdfFallback) ID() UnitID {
+	return n.PDo.UID
+}
+
+// Up performs some one-time setup and then defers to the ParDo's Up method.
+func (n *SdfFallback) Up(ctx context.Context) error {
+	sdf := (*graph.SplittableDoFn)(n.PDo.Fn)
+	addContext := func(err error) error {
+		return errors.WithContextf(err, "SdfFallback transform with UID %v", n.ID())
+	}
+	var err error
+	if n.initRestInv, err = newCreateInitialRestrictionInvoker(sdf.CreateInitialRestrictionFn()); err != nil {
+		return addContext(err)
+	}
+	if n.splitInv, err = newSplitRestrictionInvoker(sdf.SplitRestrictionFn()); err != nil {
+		return addContext(err)
+	}
+	if n.trackerInv, err = newCreateTrackerInvoker(sdf.CreateTrackerFn()); err != nil {
+		return addContext(err)
+	}
+	return n.PDo.Up(ctx)
+}
+
+// StartBundle just defers to the ParDo's StartBundle method.
```

Comment:
Editorial nits: 
We can probably remove the "just".
Given "defer" is a keyword in Go, using defers might be misinterpreted. Consider replacing it with "calls"

```suggestion
// StartBundle calls the ParDo's StartBundle method.
```

Reply:
Done, for all the comments in this file.